metricflow/model/validations/validator_helpers.py

```diff
@@ -147,13 +156,20 @@ def level(self) -> ValidationIssueLevel:
 
         raise NotImplementedError
 
-    def as_readable_str(self, with_level: bool = True) -> str:
+    @classmethod
+    def build_validation_header_msg(cls, level: ValidationIssueLevel) -> str:
+        """Builds the header message with colour."""
+        return click.style(level.name, bold=True, fg=ISSUE_COLOR_MAP[level])
+
+    def as_readable_str(self) -> str:
         """Return a easily readable string that can be used to log the issue."""
-        msg_base = f"{self.level.name} " if with_level else ""
-        msg_base += self.context.context_str() if self.context else ""
-        if msg_base:
-            msg_base += " - "
-        return msg_base + self.message
+
+        # The following is two lines instead of one line because
+        # technically self.context.context_str() can return an empty str
+        context_str = self.context.context_str() if self.context else ""
+        context_str += " - " if context_str != "" else ""
+
+        return f"{self.build_validation_header_msg(self.level)}: {context_str}{self.message}"
```

Comment:
nit: Is this intentional? ` build_validation_header_msg` is a classmethod.

Reply:
Intentional that it's being called with `self.`? My understanding is that class methods can be called from instances. Given that we had the self referential instance, I figured why not use it. But I have no preference. I can see how actually using `ValidationIssue.` instead of `self.` would be more clear. Will switch it up.